package/src/memmon.cpp

```diff
@@ -135,3 +137,44 @@ void const memmon::get_unit_info(nlohmann::json& unit_json) {
   prmon::fill_units(unit_json, params);
   return;
 }
+
+// Toggle on fast memmory monitoring
+void const memmon::do_fastmon() {
+  // Fast monitoring reads the data from a special file
+  // This file is called smaps_rollup instead of smaps
+  input_filename = "smaps_rollup";
+
+  // First discover all available metrics by peeking into self smaps_rollup file
+  std::set<std::string> available_metrics;
+  std::stringstream smaps_fname{"/proc/self/smaps_rollup"};
+  std::ifstream smap_stat{smaps_fname.str()};
+  std::string key_str{}, value_str{};
+  while (smap_stat) {
+    smap_stat >> key_str >> value_str;
+    smap_stat.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    if (smap_stat) {
+      if (key_str == "Size:") {
+        available_metrics.insert("vmem");
+      } else if (key_str == "Pss:") {
+        available_metrics.insert("pss");
+      } else if (key_str == "Rss:") {
+        available_metrics.insert("rss");
+      } else if (key_str == "Swap:") {
+        available_metrics.insert("swap");
+      }
+    }
+  }
+
+  // Delete unavailable metrics from the monitored stats
+  // In C++17/20 there are more elegant ways to do this
+  for (auto it = mem_stats.cbegin(); it != mem_stats.cend();) {
+    // Delete unavailable metrics
+    if (available_metrics.count(it->first) == 0) {
+      spdlog::warn("Metric " + it->first +
```

Comment:
Shouldn't this be "info" instead of "warn"? Warn seems a bit hard when the user actually asked for it (`--fast-memmon`)

Reply:
The logic behind the warning was that although the user asked for fast monitoring it might be unbeknownst to them that certain metrics might be missing in that configuration (although we can document this in the README etc.). Having said that, info is also OK w/ me.

Reply:
I am just wary of forcing warnings that will just be ignored (cf. ATLAS reconstruction jobs in Run 1!)

Reply:
Believe it or not we had the same discussion just two days ago 🤣  History repeats itself. Let's print this as an info message and I'll add some text to README.